### ISSUE TITLE
Orchard admin : status and todo

### ISSUE THREAD

**Issue by Skrypt** [OrchardCMS/Orchard#7016]
Here is a list of the core modules that still need to have adjustments for making them act better on mobile devices : 

- Widgets
    - needs a better UI
- Navigation
    - needs a better UI
- Workflows
    - needs some tweaks only
- Media gallery
    - needs a better UI
- Tenants
    - needs a better UI
- Modules/Features
    - needs some tweaks only

Here are the things that needs to be improved generally in all views : 

- File uploads controls
- buttons ".btn" needs to use icons
- every form controls must have the ".form-control" class
- every buttons must be big enough to be pushed by a finger
- tables must be inside a <div class="table-responsive"> to make them overflow

**Comment by sfmskywalker** [comment]
Contents -> Improve LocalizationPart rendering in SummaryAdmin mode:
![image](https://cloud.githubusercontent.com/assets/938393/16531560/440d29d6-3fcd-11e6-9228-3532c13c9f88.png)

**Comment by Skrypt** [comment]
We could replace the Navigation UI tree jquery plugin with an Angular one.
https://github.com/angular-ui-tree/angular-ui-tree

**Comment by sfmskywalker** [comment]
I actually prefer to stick with what comes with Amaretti as much as possible as that makes it easier to apply updates. Unless there is a compelling reason to switch to Angular UI tree?

**Comment by Skrypt** [comment]
Amaretti uses jquery.nestable
https://github.com/dbushell/Nestable
It's not supported by the owner of the repository. That's mainly why. Other than that I did not have time to make a comparison of features between both.
At first sight I would say that I prefer the styling of the UI tree in the Amaretti theme.
I like that when you take an element it puts a dashed box where there element was. The Angular UI Tree will higlight the zone of all the grouped elements that you are moving too. It's been a while that I've looked at that view code so I don't know if Angular would be too complex for nothing on that one. I just thought Angular would be cleaner.

**Comment by Skrypt** [comment]
We need to improve the responsive behavior of the top menu. Right now the only the "culture picker" and the "user menu" appears in it. It misses "dashboard" and "content definition" menus. I believe that the culture picker should not appear on that "hamburger" menu but should appear on the top right green portion of the header. Same goes for the user menu. Then the 2 other top left menus should appear in the "hamburger" menu.

**Comment by Skrypt** [comment]
After working a little on the top menu I think we will need to be able to display the culture picker and the user menu in that "hamburger" menu. There is not enough space left on the top green portion of the header when we are on smart phone devices.

**Comment by Skrypt** [comment]
We need to do a better javascript validation on the email test form. Right now I used the new Toastr plugin for displaying validation errors but I should not have done it that way. Reason is that we can now spam that send button to get notifiers stacking one on top of the other.

**Comment by agriffard** [comment]
I moved the Localization to Meta properties in Summary admin.

**Comment by Skrypt** [comment]
Need to refactor the implementation of panels by inheriting them instead of overriding them. See orchard-panels.less. We need to have our own panels and inherit them for reusability of the Amaretti panels.
```CSS
.or-main-panel{
  .panel;
  /* our customizations */
}
```

**Comment by Skrypt** [comment]
Merged the latest dev in the new admin theme branch

**Comment by Skrypt** [comment]
Created a new token picker that has no dependencies on JQuery UI. I've not implemented autocomplete because most of the time the textbox that has a token picker will already have a text which makes the autocomplete list filtering useless. I used a Boostrap button dropdown with some tweaks. That way we can make the pop-up box responsive and we get rid of the bug that adds a text in the bottom of the view that indicates that we can use arrows to navigate the token picker values. Feel free to look at it on the autoroute part settings.

![image](https://cloud.githubusercontent.com/assets/3228637/17686176/4094dbc6-6338-11e6-98da-5d39767838df.png)

**Comment by Skrypt** [comment]
I needed to add a validation if there was submenu items on that function so that it not apply the "open" class to a root menu item that has no submenu.

```javascript
        function openSubMenu(el, event, click) {
            var li = $(el).parent();
            var orScroller = li.find(".or-scroller");
            var subMenu = li.find("> ul");
            var isOpen = li.hasClass('open');

            //Show sub menu element
            if (!$.isXs()) {
                $("ul.visible", leftSidebar).removeClass("visible").parent().removeClass('open');

                //Add classes to create a delay on sub menu close action after mouse leave
                if (subMenu.length > 0) {
                    if (!click || (click && !isOpen && subMenu)) {
                        li.addClass('open');
                        subMenu.addClass("visible");
                    }
                    subMenu.removeClass("hide");
                }
            }
```

**Comment by Skrypt** [comment]
@Piedone I would need you to provide me with a Lombiq module that I could use for backward compatibility tests.

**Comment by Piedone** [comment]
@Skrypt could you elaborate please? I presume it should have admin features, but anything special? (BTW all of them are here: https://github.com/Lombiq/).

**Comment by Skrypt** [comment]
Yes, I need one that uses admin views. I've had issues installing modules so I need one that works with the dev branch if you can point me on a specific one that would help.

**Comment by Piedone** [comment]
Hmm. These are some modules having admin functionality and are easy to test:

- Adds some custom textboxes under a tab under Themes: https://github.com/Lombiq/Orchard-Theme-Override (needs https://github.com/Lombiq/Orchard-Ace-Editor).
- Adds a custom site settings group with textfields and checkboxes: https://github.com/Lombiq/Orchard-Azure-Application-Insights (needs https://github.com/Lombiq/Helpful-Libraries).
- Adds a tab under Users: https://github.com/Lombiq/Orchard-Login-as-Anybody

Always use the version from the dev branches.

Let me know if something doesn't look right.

**Comment by Skrypt** [comment]
The checkboxes are not displaying in some places after upgrading the Amaretti Theme. The reason is that we need absolutely to have the checkbox input right before it's label. In some places there is more than one input in the <div class="or-checkbox"> and that makes the checkbox disappear. I tried finding where this second input comes from and so far no clue.

**Comment by Skrypt** [comment]
Just to summarize this feature TODO. We have 2 bugs left to fix. We need to test if old modules still works. And we need to figure out if RTL still works. Will revert my current work on Content Items list header just to get a clean working version in the feature-admin branch. I'm stopping adding things in this branch. We need to ship it and I'm getting short on time. So please, feel free to help on the remaining tasks :wink:

**Comment by Skrypt** [comment]
I reverted back the work on content list filters. Now the filters works as before. I also added some minor responsive styling.

**Comment by Skrypt** [comment]
Culture picker is now added to the top hamburger menu. Sipke look at the background color of those icon-nav pop-up. They look different than on the Amaretti demo website.

**Comment by Skrypt** [comment]
@rtpHarry please use the admin label on your issues so that we can filter those issues directly in the issues list. And thanx for your help. It's always good to have people to test/help and have second view on this.

**Comment by rtpHarry** [comment]
@Skrypt I don't have that level of access to the repo. I'm just a "docs team" member so I can't do tags or manage issues on the main repo
